CMakeLists.txt

```
@@ -0,0 +1,110 @@
+  cmake_policy(SET CMP0083 NEW)
+  include(CheckPIESupported)
+  check_pie_supported(OUTPUT_VARIABLE check_pie_output)
+  if(NOT CMAKE_CXX_LINK_PIE_SUPPORTED)
+    list(APPEND configure_warnings "PIE link options are not supported for executable targets: ${check_pie_output}.")
+  endif()
+else()
+  list(APPEND configure_warnings "No PIE options will be passed to a linker for executable targets.")
+endif()
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+add_subdirectory(src)
+
+message("\n")
+message("Configure summary")
+message("=================")
+get_directory_property(definitions COMPILE_DEFINITIONS)
+string(REPLACE ";" " " definitions "${definitions}")
+message("Preprocessor defined macros ........... ${definitions}")
+message("C compiler ............................ ${CMAKE_C_COMPILER}")
+message("CFLAGS ................................ ${CMAKE_C_FLAGS}")
+message("C++ compiler .......................... ${CMAKE_CXX_COMPILER}")
+message("CXXFLAGS .............................. ${CMAKE_CXX_FLAGS}")
+get_directory_property(common_compile_options COMPILE_OPTIONS)
+string(REPLACE ";" " " common_compile_options "${common_compile_options}")
+message("Common compile options ................ ${common_compile_options}")
+get_directory_property(common_link_options LINK_OPTIONS)
+string(REPLACE ";" " " common_link_options "${common_link_options}")
+message("Common link options ................... ${common_link_options}")
+if(DEFINED CMAKE_BUILD_TYPE)
+  message("Build type:")
+  message(" - CMAKE_BUILD_TYPE ................... ${CMAKE_BUILD_TYPE}")
+  string(TOUPPER "${CMAKE_BUILD_TYPE}" build_type)
+  message(" - CFLAGS ............................. ${CMAKE_C_FLAGS_${build_type}}")
+  message(" - CXXFLAGS ........................... ${CMAKE_CXX_FLAGS_${build_type}}")
+  message(" - LDFLAGS for executables ............ ${CMAKE_EXE_LINKER_FLAGS_${build_type}}")
+  message(" - LDFLAGS for shared libraries ....... ${CMAKE_SHARED_LINKER_FLAGS_${build_type}}")
+else()
+  message("Available configurations .............. ${CMAKE_CONFIGURATION_TYPES}")
+  message("Debug configuration:")
+  message(" - CFLAGS ............................. ${CMAKE_C_FLAGS_DEBUG}")
+  message(" - CXXFLAGS ........................... ${CMAKE_CXX_FLAGS_DEBUG}")
+  message(" - LDFLAGS for executables ............ ${CMAKE_EXE_LINKER_FLAGS_DEBUG}")
+  message(" - LDFLAGS for shared libraries ....... ${CMAKE_SHARED_LINKER_FLAGS_DEBUG}")
+  message("Release configuration:")
+  message(" - CFLAGS ............................. ${CMAKE_C_FLAGS_RELEASE}")
+  message(" - CXXFLAGS ........................... ${CMAKE_CXX_FLAGS_RELEASE}")
+  message(" - LDFLAGS for executables ............ ${CMAKE_EXE_LINKER_FLAGS_RELEASE}")
+  message(" - LDFLAGS for shared libraries ....... ${CMAKE_SHARED_LINKER_FLAGS_RELEASE}")
+endif()
+message("\n")
+if(configure_warnings)
+    message("  ******\n")
+    foreach(warning IN LISTS configure_warnings)
+      message("  WARNING: ${warning}\n")
```

nit:

```suggestion
      message(WARNING "${warning}")
```

https://cmake.org/cmake/help/latest/command/message.html

---

Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/27060#issuecomment-1429846434).